libsolutil/JSON.h

```diff
@@ -67,4 +67,67 @@ std::string jsonPrint(Json::Value const& _input, JsonFormat const& _format);
 /// \return \c true if the document was successfully parsed, \c false if an error occurred.
 bool jsonParseStrict(std::string const& _input, Json::Value& _json, std::string* _errs = nullptr);
 
+namespace detail
+{
+
+template<typename T>
+struct helper;
+
+#define DEFINE_HELPER(TYPE, CHECK_TYPE, CONVERT_TYPE)                                                     \
+	template<>                                                                                            \
+	struct helper<TYPE>                                                                                   \
+	{                                                                                                     \
+		static bool ofType(Json::Value const& _input, std::string const& _name)                           \
+		{                                                                                                 \
+			return _input[_name].CHECK_TYPE();                                                            \
+		}                                                                                                 \
+		static TYPE get(Json::Value const& _input, std::string const& _name)                              \
+		{                                                                                                 \
+			return _input[_name].CONVERT_TYPE();                                                          \
+		}                                                                                                 \
```

Comment:
Yeah, these functions return copies. But I think it is ok in this context.

Reply:
If they return copies then it's fine.

It would be bad if they were returning references and your function was converting them to copies. If they're already copies (and will be automatically moved) then it's not an issue.